Improve PyAPI loading of cross sections from HDF5

smharper opened the pull request:
This PR makes some minor changes to the recently merged HDF5 data features.
* Fix bug where reaction thresholds were not used
* Automatically build summed reactions
* Allow Tab1 evaluations of un-ordered energies

paulromano commented on openmc/data/neutron.py:
```diff
@@ -326,6 +326,17 @@ def from_hdf5(cls, group_or_filename):
 
+        # Build summed reactions.  Start from the highest MT number because high
+        # MTs never depend on lower MTs.
+        for mt_sum in sorted(SUM_RULES.keys())[::-1]:
```
```python
sorted(..., reverse=True)
```

paulromano replied:
And actually, sorting on a dictionary uses key by default so you should be able to just do
```Python
sorted(SUM_RULES, reverse=True)
```

paulromano commented on openmc/data/reaction.py:
```diff
@@ -400,7 +400,7 @@ def from_hdf5(cls, group, energy):
         if 'xs' in group:
             xs = group['xs'].value
-            rx.xs = Tabulated1D(energy, xs)
+            rx.xs = Tabulated1D(energy[rx.threshold_idx:], xs)
```
Ooo, nice catch

smharper commented:
As discussed in person, I modified the notebook showing how to call the `xs` functions which works even for summed reactions.  I had to do some re-ordering to make it fit well, and then I added a little more content.  Have a look and let me know if should be focused more tightly on the `from_ace` side of the data module.

paulromano commented:
Thanks @smharper! I like the changes you made in the notebook.
